Magento_Theme/web/js/custom-modal.js

```diff
@@ -0,0 +1,44 @@
+define(['jquery', 'jquery-ui-modules/widget', 'mage/cookies'], function ($) {
+  $.widget('vaimo.modalWidget', {
+    options: {
+      modalOpenClass: 'modal',
+      modalNotOpenClass: 'not-open-modal',
+      modalShadowClass: 'modal-shadow',
+    },
+
+    _create: function () {
+      if (!$.mage.cookies.get('modalClosed')) {
```

Comment:
in the case that you use the cookie name in different methods, it'd be better to move it to options and reuse only as a variable